cli/cmd/install.go

```diff
@@ -71,54 +74,24 @@ type (
 		PublicAPIResources,
 		SPValidatorResources,
 		TapResources,
-		WebResources *resources
+		WebResources *charts.Resources
 
 		Identity         *installIdentityValues
 		ProxyInjector    *proxyInjectorValues
 		ProfileValidator *profileValidatorValues
 		Tap              *tapValues
+		Proxy            *proxyValues
+		ProxyInit        *proxyInitValues
 	}
 
 	configJSONs struct{ Global, Proxy, Install string }
 
-	resources   struct{ CPU, Memory constraints }
-	constraints struct{ Request, Limit string }
-
-	installIdentityValues struct {
-		Replicas uint
-
-		TrustDomain     string
-		TrustAnchorsPEM string
-
-		Issuer *issuerValues
-	}
-
-	issuerValues struct {
-		ClockSkewAllowance string
-		IssuanceLifetime   string
-
-		KeyPEM, CrtPEM string
-
-		CrtExpiry time.Time
-
-		CrtExpiryAnnotation string
-	}
-
-	proxyInjectorValues struct {
-		*tlsValues
-	}
-
-	profileValidatorValues struct {
-		*tlsValues
-	}
-
-	tapValues struct {
-		*tlsValues
-	}
-
-	tlsValues struct {
-		KeyPEM, CrtPEM string
-	}
+	installIdentityValues  charts.Identity
+	proxyInjectorValues    charts.ProxyInjector
+	profileValidatorValues charts.ProfileValidator
+	tapValues              charts.Tap
+	proxyValues            charts.Proxy
+	proxyInitValues        charts.ProxyInit
```

Comment:
curious: why redefine these types rather than using the `charts.*` variants?

Reply:
Initially, I did this with `cli.installIdentityValues` to avoid having to change the signature of multiple install and upgrade unexported functions. I can take a look again to see if it makes sense to removed these types.